src/main/java/org/assertj/core/api/ComparisonBasedAssert.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import java.math.BigDecimal;
+
+/**
+ * Assertion methods based on comparison. The implementing class should define how the comparison is performed.
+ * 
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a href="http://bit.ly/1IZIRcY"
+ *          target="_blank">Emulating
+ *          'self types' using Java Generics to simplify fluent API implementation</a>&quot; for more details.
+ * @param <ACTUAL> the type of the "actual" value.
+ * 
+ * @author Stefano Cordio
+ * @since 3.19.0
+ */
+public interface ComparisonBasedAssert<SELF extends ComparisonBasedAssert<SELF, ACTUAL>, ACTUAL> {
```

Comment:
The javadoc in this interface has to be adjusted (currently copy-pasted from `ComparableAssert`)